lib/rex/post/meterpreter/ui/console/command_dispatcher/extapi/clipboard.rb

```diff
@@ -365,52 +374,64 @@ def cmd_clipboard_monitor_stop(*args)
         download_images = val.downcase != 'false'
       when "-f"
         download_files = val.downcase != 'false'
+      when '--force'
+        force_overwrite = true
       when "-h"
         print_clipboard_monitor_stop_usage
         return true
       end
     }
+    
 
     dump = client.extapi.clipboard.monitor_stop({
       :dump           => dump_data,
       :include_images => download_images
     })
 
-    parse_dump(dump, download_images, download_files, download_path) if dump_data
+    parse_dump(dump, download_images, download_files, download_path, force_overwrite) if dump_data
 
     print_good("Clipboard monitor stopped")
   end
 
 private
 
-  def download_file( dest_folder, source )
+  def download_file( dest_folder, source, force_overwrite=false )
     stat = client.fs.file.stat( source )
     base = ::Rex::Post::Meterpreter::Extensions::Stdapi::Fs::File.basename( source )
+    attempted_overwrite = false
 
     # Basename ends up with a single name/folder. This is the only point where it
     # may be possible to do a dir trav up one folder. We need to check to make sure
     # that the basename doesn't result in a traversal
-    return false if base == '..'
-
-    dest = File.join( dest_folder, base )
+    return false, attempted_overwrite if base == '..'
 
+    local_dest_path = File.join( dest_folder, base )
+    local_dest_path = ::File.expand_path(local_dest_path)
+    
+    return false, attempted_overwrite unless local_dest_path.start_with?(::File.expand_path(dest_folder)+'/')
```

Comment:
While I haven't tested the code yet, I've reviewed everything. This is the only thing I see that needs adjustment:
```suggestion
    return false, attempted_overwrite unless local_dest_path.start_with?(::File.expand_path(dest_folder) + ::File::SEPARATOR)
```

Basically we want to use `::File::SEPARATOR` so we're making sure that we're using the local file systems desired separator vs hard coding it to `'/'`.

Reply:
Right, thanks for noticing! Should be fixed